File: Easy/credit-card-verifier-luhns-algorithm/credit-card-verifier-luhns-algorithm.cs
using System;
using System.Linq;
using System.IO;
using System.Text;
using System.Collections;
using System.Collections.Generic;


class Solution
{
    static void Main(string[] args)
    {
        int n = int.Parse(Console.ReadLine());
        for (int i = 0; i < n; i++)
        {
            string card = Console.ReadLine();
            Verifier(card);
        }
    }

    static void Verifier(string card)
    {
        card = card.Replace(" ", "");
        int secondStep = 0;
        for (int i = 0; i < card.Length; i += 2)
        {
            int factor = ((int) card[i]-48) * 2;
            secondStep += factor < 10 ? factor : factor - 9;
        }
        int thirdStep = 0;
        for (int i = 1; i < card.Length; i += 2)
        {
            thirdStep += (int) card[i]-48;
        }
        int fourthStep = secondStep + thirdStep;
        Console.WriteLine(fourthStep % 10 == 0 ? "YES" : "NO");
    }
}

File: Easy/credit-card-verifier-luhns-algorithm/credit-card-verifier-luhns-algorithm.js
const verifier = (card) => {
    card = card.replace(/ /g, "").split("").map(Number);
    let secondStep = 0;
    for (let i = 0; i < card.length; i+=2) {
        const factor = card[i] * 2;
        secondStep += factor < 10 ? factor : factor - 9;
    }
    let thirdStep = 0;
    for (let i = 1; i < card.length; i+=2) {
        thirdStep += card[i];
    }
    let fourthStep = secondStep + thirdStep;
    console.log(fourthStep % 10 === 0 ? "YES" : "NO");
};

const n = parseInt(readline());
for (let i = 0; i < n; i++) {
    const card = readline();
    verifier(card);
}